lib/node_modules/@stdlib/ndarray/concat1d/lib/main.js

```diff
@@ -0,0 +1,127 @@
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var broadcastScalar = require( '@stdlib/ndarray/base/broadcast-scalar' );
+var concat = require( '@stdlib/ndarray/concat' );
+var ndims = require( '@stdlib/ndarray/base/ndims' );
+var isArrayLikeObject = require( '@stdlib/assert/is-array-like-object' );
+var isndarrayLike = require( '@stdlib/assert/is-ndarray-like' );
+var broadcastArray = require( '@stdlib/ndarray/broadcast-array' );
+var format = require( '@stdlib/string/format' );
+var defaults = require( '@stdlib/ndarray/defaults' );
+var resolveDataTypes = require( './resolve_dtypes.js' );
+var resolveOrder = require( './resolve_order.js' );
+
+
+// VARIABLES //
+
+var DEFAULT_ORDER = defaults.get( 'order' );
+var DEFAULT_DTYPE = defaults.get( 'dtypes.default' );
+
+
+// MAIN //
+
+/**
+* Returns a one-dimensional ndarray formed by concatenating the provided input arguments.
+*
+* @param {...*} arguments - input arguments
+* @throws {Error} function must be provided an array of inputs or more than one separate input argument
+* @throws {RangeError} function must be provided a one-dimensional ndarray or a zero-dimensional ndarray
+* @returns {ndarray} output ndarray
+*
+* @example
+* var array = require( '@stdlib/ndarray/array' );
+* var ndarray2array = require( '@stdlib/ndarray/to-array' );
+*
+* var x = array( [ -1.0, 2.0, 3.0, 4.0 ] );
+* var y = array( [ -5.0, 6.0, -7.0, -8.0, 9.0, -10.0 ] );
+*
+* var out = concat1d( x, y );
+* // returns <ndarray>
+*
+* var arr = ndarray2array( out );
+* // returns [ -1.0, 2.0, 3.0, 4.0, -5.0, 6.0, -7.0, -8.0, 9.0, -10.0 ]
+*/
+function concat1d() {
+	var nargs;
+	var args;
+	var arrs;
+	var ord;
+	var dt;
+	var i;
+	var d;
+
+	nargs = arguments.length;
+
+	// Resolve function arguments...
+	if ( nargs < 1 ) {
+		throw new Error( format( 'invalid argument. The function must be provided an array of inputs or more than one separate input argument. Value: `%s`.', nargs ) );
+	}
+	if ( nargs > 1 ) {
+		args = [];
+		for ( i = 0; i < nargs; i++ ) {
+			args.push( arguments[ i ] );
+		}
+	}
+	if ( nargs === 1 ) {
+		if ( isArrayLikeObject( arguments[ 0 ] ) ) {
+			args = arguments[ 0 ];
+		} else {
+			args = [];
+			args.push( arguments[ 0 ] );
+		}
+	}
+	// Resolve the `dtype` and `order` of input ndarrays:
+	arrs = [];
+	for ( i = 0; i < args.length; i++ ) {
+		if ( isndarrayLike( args[ i ] ) ) {
+			d = ndims( args[ i ] );
+			if ( d > 1 ) {
+				throw new RangeError( format( 'invalid argument. The function must be provided a one-dimensional ndarray or a zero-dimensional ndarray. Value: `%s`.', d ) );
+			}
+			arrs.push( args[ i ] );
+		}
+	}
+	if ( arrs.length >= 1 ) {
+		dt = resolveDataTypes( arrs );
+		ord = resolveOrder( arrs );
+	} else {
+		dt = DEFAULT_DTYPE;
+		ord = DEFAULT_ORDER;
+	}
+	// Broadcast scalar or 0d ndarray inputs:
+	for ( i = 0; i < args.length; i++ ) {
+		if ( isndarrayLike( args[ i ] ) ) {
```

Comment:
This is the second time you are checking whether `args[ i ]` is an ndarray. It would be better if we can avoid what is potentially an expensive check. Ditto for the `ndims()` call below. E.g., in the L95 `for` loop, you could create additional internal arrays: `isnd` and `dims`. Then, here, you could do `if ( isnd[ i ] ) {....}` and `if ( dims[ i ] === 0 ) {...}`.

In short, think about how you can avoid potentially costly calls.